src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -279,18 +279,26 @@ def extra_resources(self, compile_context):
 
     return result
 
-  def extra_resources_digest(self, compile_context):
-    """Compute a Digest for the extra_resources for the given context."""
+  def extra_post_compile_resources_digest(self, compile_context):
+    """Compute a Digest for the post_compile_extra_resources for the given context."""
     # TODO: Switch to using #7739 once it is available.
-    extra_resources = self.extra_resources(compile_context)
+    extra_resources = self.post_compile_extra_resources(compile_context)
     if not extra_resources:
       return EMPTY_DIRECTORY_DIGEST
+
     with temporary_dir() as tmpdir:
+      rel_post_compile_merge_dir = fast_relpath(compile_context.post_compile_merge_dir, get_buildroot())
+      root_dir = os.path.join(tmpdir, rel_post_compile_merge_dir)
+      safe_mkdir(root_dir)
```

Comment:
Confusingly (it has a `makedirs` arg), `safe_file_dump` will always create parent directories, because `safe_open` does.

Could deprecate that arg, probably.

Reply:
Ack.

Also I realized that by changing from `<tmpdir>/` to `<tmpdir>/<post_merge_dir>/` would probably break javac apt. so will add a test to it.

Reply:
tests added